pkg/resourcecollector/persistentvolumeclaim.go

```diff
@@ -101,11 +102,23 @@ func (r *ResourceCollector) preparePVCResourceForApply(
 		// In the case of storageClassMappings, we need to reset the
 		// storage class annotation and the provisioner annotation
 		var newSc string
+		var currentSc string
 		var exists bool
 		var provisioner string
-		if val, ok := pvc.Annotations[v1.BetaStorageClassAnnotation]; ok {
-			if newSc, exists = storageClassMappings[val]; exists && len(newSc) > 0 {
-				pvc.Annotations[v1.BetaStorageClassAnnotation] = newSc
+		// Get the existing storage class from the pvc spec
+		// It can be in BetaStorageClassAnnotation annotation or in the spec.
+		currentSc, err := utils.GetStorageClassNameForPVC(&pvc)
```

Comment:
Lets not error out here as well if we don't find a storage class.

Reply:
This case will be executed only when storageclass mapping is present / passed from caller. In this case, storage class will be always present. So returning error